subsys/bluetooth/host/Kconfig

```diff
@@ -351,6 +351,24 @@ config BT_OOB_DATA_FIXED
 	  value. This option should only be enabled for debugging and should
 	  never be used in production.
 
+config BT_KEY_DISPLACEMENT
+	bool "Displace oldest keys with new ones if key storage is full"
+	help
+	  With this option enabled, if a pairing attempt occurs and the key storage
+	  is full, then the oldest keys in storage will be displaced to free space
+	  for the new pairing keys.
+
+if BT_KEY_DISPLACEMENT
+config BT_KEY_DISPLACEMENT_CNTR_IN_SETTINGS
```

Comment:
I'm not sure we need this separate option. For other submodules we simply use `BT_SETTIGS` to decide if something should be stored in flash or not, i.e. the condition is in practice `#if defined(CONFIG_BT_SOME_FEATURE) && defined(CONFIG_BT_SETTIGNS)`

Reply:
Defining BT_KEY_DISPLACEMENT_CNTR_IN_SETTINGS will break compatibility with records stored before this feature enabled ind slightly increase storage size.

Reply:
Yes, though I don't see how that's related to my suggestion. You'll still need to explicitly enable `BT_KEY_DISPLACEMENT` (or whatever other name we decide for it) to break the compatibility. Also, we already now have this issue e.g. if you toggle `BT_SMP_SC_ONLY` it will change `sizeof(struct bt_keys)`.

Reply:
If I do not enable BT_KEY_DISPLACEMENT_CNTR_IN_SETTINGS but enable BT_KEY_DISPLACEMENT compatibility is not broken and counters are initialized during loading process each time but not stored

Reply:
I understand, but why would someone want to enable the feature if they don’t get persistency, yet have it for all other info?

Reply:
If not saved in settings explicitly, counter will be recovered during settings load process when a storage is allocated from the keys_set() function. This works only if settings backend does not change record order.

Reply:
@Nick0las-kipt ok, that makes kind of sense. So for something like FCB we wouldn't need the counter in flash then? Although I suppose even with FCB it will change if you go and do e.g. an encryption key refresh procedure at some point in the future. Anyway, if we keep this, would something like `<prefix>_EXPLICIT_ORDERING` make sense as the Kconfig variable, since we're explicitly enforcing it rather than the ordering coming implicitly with the help of the settings backend?

Reply:
@Nick0las-kipt, @jhedberg, due to the fcb compression algorithm the order will change unless all keys are in the same sector.